localstack-core/localstack/services/cloudformation/engine/v2/change_set_model_preproc.py

```diff
@@ -908,6 +916,35 @@ def visit_node_parameter(self, node_parameter: NodeParameter) -> PreprocEntityDe
         before = dynamic_delta.before or default_delta.before
         after = dynamic_delta.after or default_delta.after
 
+        # handle dynamic references, e.g. references to SSM parameters
+        # TODO: support more parameter types
+        parameter_type: str = node_parameter.type_.value
+        if parameter_type.startswith("AWS::SSM"):
+            if parameter_type in [
+                "AWS::SSM::Parameter::Value<String>",
+                "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
+                "AWS::SSM::Parameter::Value<CommaDelimitedList>",
+            ]:
+                if is_nothing(after):
+                    if is_nothing(before):
+                        raise RuntimeError(
+                            "Invalid state: no resolved parameter found for SSM parameter {node_parameter.name}"
+                        )
+                    # we don't need to resolve the parameter here since it was done in the previous deployment
+                    after = before
+                else:
+                    resolved_value = resolve_ssm_parameter(
+                        account_id=self._change_set.account_id,
+                        region_name=self._change_set.region_name,
+                        stack_parameter_value=after,
+                    )
+                    after = ResolvedParameter(value=after, resolved_value=resolved_value)
```

Comment:
why are we not resolving the resolved parameter here (this visitor function), but later?